rest/indextest/index_init_api_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2025-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package indextest
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	sgbucket "github.com/couchbase/sg-bucket"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/db"
+	"github.com/couchbase/sync_gateway/rest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestChangeIndexPartitions ensures that users can change the number of partitions without any downtime using the documented steps.
+//  1. Have a database already running with an undesired number of partitions.
+//  2. Use Index Init API to initialize a new set of indexes with different partitions.
+//  3. Use the regular db config API to change the number of partitions to match the set created.
+//  4. Ensure the database starts up using the indexes created in step 2 without any delay.
+//  5. Use the _post_upgrade endpoint to clean up the old set of indexes.
+func TestChangeIndexPartitions(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyHTTP, base.KeyConfig, base.KeyQuery)
+	require.False(t, base.TestsDisableGSI(), "Test requires GSI to be enabled")
+	const (
+		dbName            = "db"
+		initialPartitions = uint32(2)
+		newPartitions     = uint32(4)
+	)
+
+	rt := rest.NewRestTesterPersistentConfigNoDB(t)
```

Comment:
Does this work with and without persistent config? Some of our k8s customers do not use k8s.

Reply:
it should work but I'll add a different test to verify